### ISSUE TITLE
Exception when using Serilog.Settings.Configuration

### ISSUE THREAD

**Issue by KevinGliewe** [Fody/Anotar#656]
# Description

Using [`Serilog.Settings.Configuration`](https://github.com/serilog/serilog-settings-configuration) to configure serilog will produce an exception:

```text
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Anotar.Serilog, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
File name: 'Anotar.Serilog, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.InternalLoad(ObjectHandleOnStack assemblyName, ObjectHandleOnStack requestingAssembly, StackCrawlMarkHandle stackMark, Boolean throwOnFileNotFound, ObjectHandleOnStack assemblyLoadContext, ObjectHandleOnStack retAssembly)
   at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, RuntimeAssembly requestingAssembly, StackCrawlMark& stackMark, Boolean throwOnFileNotFound, AssemblyLoadContext assemblyLoadContext)
   at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
   at Serilog.Settings.Configuration.ConfigurationReader.LoadConfigurationAssemblies(IConfigurationSection section, AssemblyFinder assemblyFinder)
   at Serilog.Settings.Configuration.ConfigurationReader..ctor(IConfigurationSection configSection, AssemblyFinder assemblyFinder, IConfiguration configuration)
   at Serilog.ConfigurationLoggerConfigurationExtensions.Configuration(LoggerSettingsConfiguration settingConfiguration, IConfiguration configuration, String sectionName, DependencyContext dependencyContext)
   at Serilog.ConfigurationLoggerConfigurationExtensions.Configuration(LoggerSettingsConfiguration settingConfiguration, IConfiguration configuration, DependencyContext dependencyContext)
   at Program.<Main>$(String[] args) in #ProjectPath#\Program.cs:line 8
```

`Serilog.Settings.Configuration` trys to load all assemblies containing the name "serilog" and referencing serilog.

Somehow [`Microsoft.Extensions.DependencyModel`](https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.DependencyModel) finds `Anotar.Serilog.Fody` and adds it to `DependencyContext.Default.RuntimeLibraries`.

# Minimal Repo

I created a minimal repo with more informations about my findings:

[Anotar.Serilog.Issue](https://github.com/KevinGliewe/Anotar.Serilog.Issue)

# Other Issue

I also created a issue on the `Serilog.Settings.Configuration` repo:
https://github.com/serilog/serilog-settings-configuration/issues/303

I created an issue on both projects, because there are two things that i find strange:

 - Why can't i exclude assemblys from loading on `Serilog.Settings.Configuration` (Not relevant here)
 - Why does [`Microsoft.Extensions.DependencyModel`](https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.DependencyModel) pick up `Anotar.Serilog.Fody` as an reference? Shouldn't Anotar remove it?

**Comment by KevinGliewe** [comment]
Ok, my bad. I found a solution:

https://github.com/serilog/serilog-settings-configuration/issues/303#issuecomment-1078122519

But one question remains: Shouldn't Anotar remove the reference in general? Or is it even possible?

**Comment by ltrzesniewski** [comment]
> Shouldn't Anotar remove the reference in general? Or is it even possible?

It should:

https://github.com/Fody/Anotar/blob/419326a14fc4d354ecad62169aed14afa4360945/Serilog/Anotar.Serilog.Fody/ModuleWeaver.cs#L27

And it does:

![image](https://user-images.githubusercontent.com/7913492/163686398-6a8b8ec1-4c56-43fe-977a-3ad217edc459.png)

The problem here is that, although the reference is removed from the dll, it is still present in the deps.json file:

```JSON
"Anotar.Serilog.Issue/1.0.0": {
  "dependencies": {
    "Anotar.Serilog.Fody": "6.0.0",
    "Fody": "6.6.0",
    "Microsoft.Extensions.Configuration": "6.0.0",
    "Serilog": "2.10.0",
    "Serilog.Settings.Configuration": "3.3.1"
  },
  "runtime": {
    "Anotar.Serilog.Issue.dll": {}
  }
},
```

And Microsoft.Extensions.DependencyModel reads the deps.json file... 😖

That's a known issue which has been present in Fody since deps.json files were introduced in .NET Core. We didn't fix it since it never was a problem (I think you're the first to report the issue). Nevertheless, it's something we should fix.

**Comment by ltrzesniewski** [comment]
I'm closing this since it's an issue in Fody, not in Anotar. You can track it at https://github.com/Fody/Fody/issues/1090.

**Comment by ltrzesniewski** [comment]
This should be fixed in Fody v6.6.1.
